### ISSUE TITLE
Google-style: Illegal to break the line before or after the lambda arrow

### ISSUE THREAD

**Issue by mohitsatr** [checkstyle/checkstyle#17253]
From: https://google.github.io/styleguide/javaguide.html#s4.5.1-line-wrapping-where-to-break

> A line is never broken adjacent to the arrow in a lambda or a switch rule, except that a break may come immediately after the arrow if the text following it consists of a single unbraced expression.

I'm still somewhat confused about what 'adjacent' could mean here. So I'm assuming rule is saying 'a line is never broken before and after the arrow in the lambda or switch, except.... '.

```java
$ cat LambdaBreak.java
/** some javadoc. */
public class LambdaBreak {

  interface MyLambdaInterface {
    int foo(int a, int b);
  };

  void test() {
    MyLambdaInterface div = (a, b)
        -> {                               // VIOLATION EXPECTED
          if (b != 0) {
            return a / b;
          }
          return 0; // warn
        };
  }

  void test2(int day) {
    ArrayList<String> list = new ArrayList<>();
    list.stream()
    .map(x 
        -> x.toString())                     // VIOLATION EXPECTED
    .forEach(System.out::println);
    
    String dayString = switch (day) {
      case 1 
          -> "one day of the week";        // VIOLATION EXPECTED 
      case 2 
          ->                                  // VIOLATION EXPECTED
              "two day of the week";
      case 3 ->                                       // OK
          "third day of the week"; 
      default ->                                      // OK
          throw new IllegalArgumentException();
    };
  }
}

```

```xml
$ java -jar checkstyle-10.25.0-all.jar -c google_checks.xml LambdaBreak.java
Starting audit...
Audit done.
```

we are not following this rule.

**Comment by romani** [comment]
rule is to violate:
```
MyLambda<String, Long, Object> lambda =
    (String label, Long value, Object obj) ->
    { // violation above or on this line
      ...
    };
```

in other words "\n" after `->` is ok, only if no "{".
if there is `{` after `->` , then `{` should be on same line as `->`

**Comment by Zopsss** [comment]
@mohitsatr please send a supplemental PR to add blue tick in the coverage table with a reference of this issue

**Comment by vivek-0509** [comment]
@romani 
I think we can enforce this rule using a combination of two existing checks. By adding TokenTypes.LAMBDA to OperatorWrapCheck's acceptable tokens and configuring it with EOL option, we can catch violations where line breaks occur before the arrow. The existing LeftCurly check (already configured with LAMBDA in google_checks.xml) handles the case where a line break occurs after the arrow followed by {. Together, these two checks fully enforce the rule while correctly allowing breaks after -> when followed by an unbraced expression. I have tested this locally and it catches all the violations from the issue description.

However, this approach requires both checks to be configured together to fully enforce the rule. If a user disables either check, the rule would only be partially enforced.

An alternative would be to create a dedicated check  (LambdaArrowWrapCheck  we can think about name) that enforces the complete rule in one place with clear, specific error messages.
What do you think we should do here ??

**Comment by vivek-0509** [comment]
@romani ping

**Comment by vivek-0509** [comment]
@romani this should be tagged as google style, working on it

**Comment by romani** [comment]
Fix is merged
